### PR TITLE
Use rake.bat when on Windows

### LINKED ISSUE #1: PATH problems on Windows
Glad to see someone's trying to get `rake` working with the build package. I have not been able to get the `build` package to work on my system. When I use it with this package, I get this error: 

``` sh
Executing: rake build 
Unable to execute: rake
Make sure `cmd` and `cwd` exists and have correct access permissions.
Build finds binaries in these folders: C:\Ruby22\bin;C:\Users\andy\AppData\Local\atom\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Java\jdk1.7.0_03\bin;C:\Program Files (x86)\Windows Live\Shared;C:\Python34;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x64;C:\Program Files (x86)\Git\bin;C:\Program Files (x86)\Git\cmd;C:\Program Files (x86)\MiKTeX 2.9\miktex\bin\;C:\Program Files (x86)\Heroku\bin;C:\Program Files (x86)\git\cmd;C:\Program Files (x86)\QuickTime\QTSystem\;
```

The `rake` command is definitely on my path - in `C:\Ruby22\bin`.

### PR DESCRIPTION
**Pull request by andymeneely**
Fixes #1

### CODE REVIEW

File: lib/build-rake.coffee [hunk 1]
```diff
@@ -14,20 +14,21 @@ exports.provideRakeBuilder = ->
...
+        rake_t    = "#{rake_exec} -T"
+        child_process.exec rake_t, {cwd: @cwd}, (error, stdout, stderr) ->
           reject(error) if error?
           config = []
           stdout.split("\n").forEach (line) ->
             if (m = /^rake (\S+)\s*#\s*(\S+.*)/.exec(line))?
               config.push
                 name: "rake #{m[1]} - #{m[2]}"
-                exec: "rake"
+                exec: "#{rake_exec}"
```
Comment (lslezak) [inline lib/build-rake.coffee]: The interpolation is not needed here, simple `exec: rake_exec` is enough.

File: spec/build-rake-spec.coffee [hunk 2]
```diff
@@ -31,24 +31,39 @@ describe 'Rakefile provider', ->
...
+          child_process.exec.mostRecentCall.args[2](null, '', '')
+
+        waitsForPromise -> builder.settings()
+
+        runs ->
+          expect(child_process.exec.mostRecentCall.args[0]).toEqual('rake -T')
+
+    describe 'on Windows', ->
+
+      it 'runs rake to get the list of tasks', ->
```
Comment (lslezak) [inline spec/build-rake-spec.coffee]: I'd describe the test as `runs rake.bat to get the list of tasks` so the difference from the previous test is more obvious.

**Comment by lslezak** [comment]
Just two minor issues, please fix them and I'll merge the pull request. The rest is OK. :+1: 

Thanks a lot!

**Comment by andymeneely** [comment]
Ok, I fixed those. I also ported the other test over to both platforms and refactored to use beforeEach.

**Comment by lslezak** [comment]
Merged, thanks a lot for the contribution!

I'll release an updated version to atom.io...

**Pull request merged**